rq/job.py

```diff
@@ -864,11 +878,19 @@ def restore(self, raw_data) -> Any:
         self.failure_ttl = int(obj.get('failure_ttl')) if obj.get('failure_ttl') else None
         self._status = obj.get('status').decode() if obj.get('status') else None
 
-        if obj.get('success_callback_name'):
+        if obj.get('success_callback_name'):  # backward compatibility
             self._success_callback_name = obj.get('success_callback_name').decode()
+        elif 'success_callback' in obj:
+            success_callback = json.loads(obj.get('success_callback').decode())
+            self._success_callback_name = success_callback['name']
+            self.success_callback_timeout = success_callback.get('timeout', CALLBACK_TIMEOUT)
 
-        if obj.get('failure_callback_name'):
+        if obj.get('failure_callback_name'):  # backward compatibility
             self._failure_callback_name = obj.get('failure_callback_name').decode()
+        elif 'failure_callback' in obj:
+            failure_callback = json.loads(obj.get('failure_callback').decode())
+            self._failure_callback_name = failure_callback['name']
+            self.failure_callback_timeout = failure_callback.get('timeout', CALLBACK_TIMEOUT)
```

Comment:
It didn't occur to me that we'd need to encode this in JSON because Redis hashes doesn't support nesting. Had I known this, I wouldn't have suggested that we store callback data in a nested structure.

Do you want to revert this to how it was previously, or do you think it's worth the trouble to serialize and deserialize to JSON? I think this just adds unneeded complexity to job persistence.